lib/perl/Genome/Model/Tools/Bsmap/MethylationRatio.pm

```diff
@@ -0,0 +1,99 @@
+package Genome::Model::Tools::Bsmap::MethylationRatio;
+
+use strict;
+use warnings;
+use Cwd;
+use Genome;
+
+use Genome::Utility::Text qw(sanitize_string_for_filesystem);
+use File::Spec;
+
+my $METHRATIO_COMMAND = File::Spec->join(getcwd, 'methylation_ratio.py');
+
+class Genome::Model::Tools::Bsmap::MethylationRatio {
+    is => 'Command',
+    has => [
+        bam_file => {
+            is => 'Text',
+            doc => 'An indexed bam file',
+            is_input => 1,
+        },
+        output_file => {
+            is => 'Text',
+            doc => 'File name for methyl counts (a .bgz extension will be added automatically)',
+            is_output => 1,
+            is_input => 1,
+            default_value => 'methylation',
+        },
+        threads => {
+            is => 'Text',
+            doc => 'Numer of threads (2 minimum: one for reading and one for writing; additional threads beyond 2 will be allocated for reading; recommended number is 5)',
+            is_output => 0,
+            is_input => 1,
+            default_value => '2',
+        },
+        output_directory => {
+            is => 'Text',
+            doc => 'Where to output the methyl counts',
+            is_output => 1,
+            is_input => 1,
+        },
+        reference => {
+            is => 'Text',
+            doc => '36, 37, or a path to the reference fasta',
+            is_input => 1,
+        },
+    ],
+};
+
+sub _reference_fasta {
+    
+    my ($self) = @_;
+
+    my %mapping = (
+        36 => 'NCBI-human-build36',
+        37 => 'GRCh37-lite-build37',
+    );
+
+    my $reference = $self->reference;
+
+    if (exists $mapping{$reference}) {
+        my $reference_build = Genome::Model::Build::ReferenceSequence->get(
+            name => $mapping{$reference}
+        );
+        $reference = $reference_build->cached_full_consensus_path('fa');
+    }
+
```

Comment:
The reference should be specified in the class definition as a Genome::Model::Build::ReferenceSequence. This also has built in erro checking, ensuring that  the id/name given is valid.

reference => {
  is => 'Genome::Model::Build::ReferenceSequence',
 is_input => 1,
 doc => ...,
}

Then getting reference fasta is easier:

sub _reference_fasta {
  my ($self) = @_;
  return $self->reference->cached_full_consensus_path('fa');
}